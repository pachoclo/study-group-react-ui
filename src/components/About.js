import React from 'react'

const About = ({ auth }) => (
  <>
    <h2>About</h2>
    {auth.isAuthenticated() && <p>(c) Study Groupssss weee</p>}
    {!auth.isAuthenticated() && <p>Not logged In!!! Please login mr/mrs...</p>}
  </>
)

export default About
