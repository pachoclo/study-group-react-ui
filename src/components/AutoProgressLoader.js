import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'

const defaults = {
  initialProgress: 5,
  intervalMillis: 500,
  increment: 5
}

/**
 * Renders a loader that automatically increments its progress
 *
 * @param {number} intervalMillis specifies how often the progress should be increased
 */
export const AutoProgressLoader = ({ intervalMillis = defaults.intervalMillis }) => {
  const [progress, setProgress] = useState(defaults.initialProgress)

  useEffect(() => {
    const intervalId = setInterval(() => {
      setProgress(currentProgress => {
        const newProgress = currentProgress + 5
        return newProgress <= 100 ? newProgress : currentProgress
      })
    }, intervalMillis)

    return function cleanUp () {
      clearInterval(intervalId)
      setProgress(100)
    }
  }, [intervalMillis])

  return (
    <div className='progress' style={{ height: '4px', backgroundColor: 'gray', borderRadius: 0 }}>
      <div
        className='progress-bar progress-bar-striped progress-bar-animated bg-warning'
        role='progressbar'
        aria-valuenow={progress}
        aria-valuemin='0'
        aria-valuemax='100'
        style={{ width: `${progress}%` }}
      />
    </div>
  )
}

AutoProgressLoader.propTypes = {
  intervalMillis: PropTypes.number
}

export default AutoProgressLoader
