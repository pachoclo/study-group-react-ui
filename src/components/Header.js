import React from 'react'
import Menu from './Menu'
import { withRouter } from 'react-router-dom'

const Header = props => <Menu {...props} />

export default withRouter(Header)
