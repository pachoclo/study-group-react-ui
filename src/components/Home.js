import React from 'react'
import { connect } from 'react-redux'
import {} from '../store/actions/authActions'

const Home = ({ isAuthenticated, profile }) => (
  <>
    <h2>Home</h2>
    {isAuthenticated && (
      <>
        <p>La casaa</p>
        <p>Profile:</p>
        <pre> {JSON.stringify(profile, null, 2)}</pre>
      </>
    )}
    {!isAuthenticated && <p>Not logged In!!! Please login mr/mrs...</p>}
  </>
)

const mapStateToProps = ({ auth }) => ({
  isAuthenticated: auth.authenticated,
  profile: auth.profile
})

export default connect(mapStateToProps)(Home)
