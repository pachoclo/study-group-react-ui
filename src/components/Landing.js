import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { preventDefault } from '../util'
import { loginRequest } from '../redux/actions/authActions'

const Landing = ({ isAuthenticated, loginRequest, loading }) => {
  return (
    <div className='landing'>
      <div className='dark-overlay landing-inner text-light'>
        <div className='container'>
          <div className='row'>
            <div className='col-md-12 text-center'>
              <h1 className='display-3 mb-4'>StudyGroup</h1>
              <p className='lead'>
                Gets you working with your peers pronto.
                <br />
                Setup meetings and organize your materials in one place
              </p>
              {!isAuthenticated && !loading && (
                <a href='login.html' className='btn btn-lg btn-info mt-5' onClick={preventDefault(loginRequest)}>
                  Login
                </a>
              )}
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = ({ auth, loader }) => ({
  isAuthenticated: auth.authenticated,
  loading: loader.loading
})

Landing.propTypes = {
  isAuthenticated: PropTypes.bool,
  loginRequest: PropTypes.func,
  loading: PropTypes.bool
}

export default connect(
  mapStateToProps,
  { loginRequest }
)(Landing)
