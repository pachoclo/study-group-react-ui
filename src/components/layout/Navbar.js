import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { loginRequest, logoutRequest } from '../../redux/actions/authActions'
import { preventDefault } from '../../util'
import { Link } from 'react-router-dom'

const Navbar = ({ isAuthenticated, loginRequest, logoutRequest, avatarUrl, nickname }) => (
  <nav className='navbar navbar-expand-sm navbar-dark bg-dark'>
    <div className='container'>
      <Link className='navbar-brand' to='/'>
        StudyGroup
      </Link>
      <button
        className='navbar-toggler'
        type='button'
        data-toggle='collapse'
        data-target='#mobile-nav'
      >
        <span className='navbar-toggler-icon' />
      </button>

      <div className='collapse navbar-collapse' id='mobile-nav'>
        <ul className='navbar-nav ml-auto'>
          {!isAuthenticated && (
            <li className='nav-item'>
              <a className='nav-link' href='/login' onClick={preventDefault(loginRequest)}>
                Login
              </a>
            </li>
          )}
          {isAuthenticated && (
            <>
              <li className='nav-item'>
                <Link className='nav-link' to='/groups'>
                  Groups
                </Link>
              </li>
              <div className='dropdown'>
                <img
                  src={avatarUrl}
                  alt={nickname}
                  className='rounded-circle avatar-img'
                  id='dropdownMenu2'
                  data-toggle='dropdown'
                  aria-haspopup='true'
                  aria-expanded='false'
                />
                <div className='dropdown-menu' aria-labelledby='dropdownMenu2'>
                  <h6 className='dropdown-header'>{nickname}</h6>
                  <div className='dropdown-divider' />
                  <a
                    className='dropdown-item'
                    href='/logout'
                    onClick={preventDefault(logoutRequest)}
                  >
                    Logout
                  </a>
                </div>
              </div>
            </>
          )}
        </ul>
      </div>
    </div>
  </nav>
)

Navbar.propTypes = {
  isAuthenticated: PropTypes.bool,
  loginRequest: PropTypes.func,
  logoutRequest: PropTypes.func,
  avatarUrl: PropTypes.string,
  nickname: PropTypes.string
}

const mapStateToProps = ({ auth }) => ({
  isAuthenticated: auth.authenticated,
  avatarUrl: auth.profile.picture,
  nickname: auth.profile.nickname
})

const actions = { loginRequest, logoutRequest }

export default connect(
  mapStateToProps,
  actions
)(Navbar)
