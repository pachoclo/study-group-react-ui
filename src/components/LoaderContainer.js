import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import AutoProgressLoader from './AutoProgressLoader'

export const LoaderContainer = ({ loading, intervalMillis }) => {
  return loading ? <AutoProgressLoader intervalMillis={intervalMillis} /> : null
}

LoaderContainer.propTypes = {
  loading: PropTypes.bool,
  intervalMillis: PropTypes.number
}

const mapStateToProps = ({ loader }) => loader

export default connect(mapStateToProps)(LoaderContainer)
