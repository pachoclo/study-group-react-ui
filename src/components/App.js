import React, { useEffect } from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { renewSession } from '../redux/actions/authActions'
import '../App.css'
import authClient from '../Auth/Auth'
import LoginCallback from './LoginCallback'
import Navbar from './layout/Navbar'
import ErrorAlert from './layout/ErrorAlert'
import Landing from './layout/Landing'
import TopLevelLoader from './TopLevelLoader'

function App ({ renewSession, error }) {
  useEffect(() => {
    // if the user hasn't logged out, try to renew the session
    authClient.isLoggedIn() && renewSession()
  })

  return (
    <Router>
      <>
        <Navbar />
        <TopLevelLoader />
        <div className='{//container}'>
          {error.err && <ErrorAlert error={error.err} followUp={error.followUp} />}
          <Route exact path='/' component={Landing} />
          <Route path='/loginCallback' component={LoginCallback} />
        </div>
      </>
    </Router>
  )
}

App.propTypes = {
  renewSession: PropTypes.func,
  error: PropTypes.object
}

const mapStateToProps = ({ error }) => ({ error })

export default connect(
  mapStateToProps,
  { renewSession }
)(App)
