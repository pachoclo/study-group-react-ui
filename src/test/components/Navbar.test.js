import React from 'react'
import { shallow } from 'enzyme'
import { Navbar } from '../../components/Navbar'

describe('Landing Test', () => {
  it('should render correctly', () => {
    expect(shallow(<Navbar />)).toMatchSnapshot()
  })

  it('should render the avatar (and menu) component if authenticated', () => {
    expect(shallow(<Navbar isAuthenticated />)).toMatchSnapshot()
  })
})
