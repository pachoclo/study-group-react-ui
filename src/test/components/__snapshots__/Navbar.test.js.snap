// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Test should render correctly 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <Link
      className="navbar-brand"
      to="/"
    >
      StudyGroup
    </Link>
    <button
      className="navbar-toggler"
      data-target="#mobile-nav"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="mobile-nav"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Landing Test should render the avatar (and menu) component if authenticated 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <Link
      className="navbar-brand"
      to="/"
    >
      StudyGroup
    </Link>
    <button
      className="navbar-toggler"
      data-target="#mobile-nav"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="mobile-nav"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            to="/groups"
          >
            Groups
          </Link>
        </li>
        <div
          className="dropdown"
        >
          <img
            aria-expanded="false"
            aria-haspopup="true"
            className="rounded-circle avatar-img"
            data-toggle="dropdown"
            id="dropdownMenu2"
          />
          <div
            aria-labelledby="dropdownMenu2"
            className="dropdown-menu"
          >
            <h6
              className="dropdown-header"
            />
            <div
              className="dropdown-divider"
            />
            <a
              className="dropdown-item"
              href="/logout"
              onClick={[Function]}
            >
              Logout
            </a>
          </div>
        </div>
      </ul>
    </div>
  </div>
</nav>
`;
